bazel/external/proxy_wasm_cpp_host.BUILD

```diff
@@ -1,4 +1,10 @@
 load("@rules_cc//cc:defs.bzl", "cc_library")
+load(
+    "@envoy//bazel:envoy_build_system.bzl",
+    "envoy_select_wasm_all_v8_wavm_none",
+    "envoy_select_wasm_v8",
+    "envoy_select_wasm_wavm",
+)
```

Comment:
This shouldn't leaked here. You should be able to just do select in `deps` where these library target are referenced. The BUILD file itself can live in proxy_wasm_cpp_host repo.

Reply:
I can have additional libraries to allow WAVM and V8 to be included separately in either external/proxy-wasm-cpp-host.BUILD or in the BUILD file in the external repository, but I am having problems with zlib:

The zlib library in envoy is strange, and if I include zlib in proxy-wasm-cpp-host and depend on it via @zlib in the deps, then envoy fails because it wants to see //external:zlib which proxy-wasm-cpp-host/BUILD doesn't like.

The envoy zlib build uses a filegroup then a net_zlib target then maps via anative.bind to zlib from @envoy/baze/foreign_cc::zlib, so there are at least 4 names involved: zlib, net_lib, @envoy/baze/foreign_cc::zlib, and //external:zlib  and it isn't clear to me which one I should be using to get the BUILD file working in envoy and in the external repo.  I have tried a couple combinations to no avail.  Any suggestion?